packages/jest-mock/src/__tests__/index.test.ts

```diff
@@ -1604,6 +1604,179 @@ describe('moduleMocker', () => {
       expect(spy2.mock.calls).toHaveLength(1);
     });
   });
+
+  describe('mockProperty', () => {
+    it('should work', () => {
+      const obj = {
+        property: 1,
+      };
+
+      const replaced = moduleMocker.mockProperty(obj, 'property', 2);
+
+      expect(obj.property).toBe(2);
+
+      replaced.mockRestore();
+
+      expect(obj.property).toBe(1);
+    });
+
+    it('should allow mocking a property multiple times', () => {
+      const obj = {
+        property: 1,
+      };
+
+      const replacedFirst = moduleMocker.mockProperty(obj, 'property', 2);
+
+      const replacedSecond = moduleMocker.mockProperty(obj, 'property', 3);
+
+      expect(obj.property).toBe(3);
+
+      replacedSecond.mockRestore();
+
+      expect(obj.property).toBe(1);
+
+      replacedFirst.mockRestore();
```

Comment:
I'm not familiar with the implementation but does a test make sense where we use `jest.restoreAllMocks` when we mocked a property multiple times?

Reply:
Tests added.